### PR TITLE
Use ESM format for import/export in lib/question.js

### PR DESCRIPTION
**Pull request by jonatanschroeder**
As discussed in our dev meeting. This is intended as a first step to simplify future changes to migrate these libs to new format. Typescript, async and changes in query functions are not done.

Lodash is still using the old notation, as it is used with chaining, and I wanted to avoid unnecessary changes in the functions themselves.

Best viewed with whitespace changes ignored.

### CODE REVIEW

File: apps/prairielearn/src/lib/question.js [hunk 1]
```diff
@@ -2,31 +2,32 @@
...
-const sqldb = require('@prairielearn/postgres');
-const questionServers = require('../question-servers');
-const workspaceHelper = require('./workspace');
-const issues = require('./issues');
-const error = require('@prairielearn/error');
-const { createServerJob } = require('./server-jobs');
-
+import * as async from 'async';
+import * as path from 'path';
+import * as jsonStringifySafe from 'json-stringify-safe';
```
Comment (nwalters512) [inline apps/prairielearn/src/lib/question.js:7]: This (and possibly others here) will be runtime errors: this code tries to call `jsonStringifySafe`, but a namespace import will *never* be directly callable. I'd also recommend installing TS types for this package.

Comment (jonatanschroeder) [reply apps/prairielearn/src/lib/question.js]: Resolved (and types installed) in 29ef820.

I also checked the other imports and it seems the others are used appropriately.

**Review by nwalters512 — APPROVED** [review]
To properly test these changes, one should use `make build && make start` instead of `make dev`, since the former is what will actually be used at runtime in production. I tested these changes that way and they appear to work after `json-stringify-safe` was fixed!

**Comment by jonatanschroeder** [comment]
> To properly test these changes, one should use `make build && make start` instead of `make dev`, since the former is what will actually be used at runtime in production.

Wouldn't CI take care of this?

**Comment by nwalters512** [comment]
> Wouldn't CI take care of this?

No, Mocha runs against a version of the code that's transpiled on the fly, which can be subtly different than what actually happens at runtime. This is a big driver for moving to ESM, where the exact same module system + resolution + behavior is used everywhere.

**Pull request merged**